pkgs/development/interpreters/python/cpython/2.7/default.nix

```diff
@@ -258,7 +258,7 @@ in with passthru; stdenv.mkDerivation ({
     LDFLAGS = lib.optionalString (!stdenv.isDarwin) "-lgcc_s";
     inherit (mkPaths buildInputs) C_INCLUDE_PATH LIBRARY_PATH;
 
-    NIX_CFLAGS_COMPILE = optionalString stdenv.isDarwin "-msse2"
+    NIX_CFLAGS_COMPILE = optionalString (stdenv.isDarwin && stdenv.isx86_64) "-msse2"
```

Comment:
```suggestion
    NIX_CFLAGS_COMPILE = optionalString (stdenv.hostPlatform.system == "x86_64-darwin") "-msse2"
```

Reply:
Longer expression but less conditionals and hence easier to read.

Reply:
Amended, but with `targetPlatform` instead of `hostPlatform` (I think that's right).

Reply:
I think it's actually `hostPlatform`. An example if you cross compile a gcc on x86_64 for aarch64, that can compile to riscv64 than: build == x86_64, host == aarch64 and target == riscv64.

but @matthewbauer can correct me here.